feat automated deployment

Saidmuhammad2002 opened the pull request:
### What was done?
Configured automated deployment of frontend app via AWS CDK.

### More detailed:

1. Created an S3 bucket, restricted to CloudFront access only.

3. Configured automatic upload of built files to S3.
5. Set up a CloudFront distribution.
7. Implemented automated CloudFront cache invalidation after deployment.
9. Added npm scripts for building the app, synthesizing the CloudFormation template, and deploying.
11. CloudFront URL is publicly accessible.
13. S3 website is restricted to CloudFront only.
15. Single npm script for building & deploying.

DeployWebAppStack.deploymentBucketNameA59FA865 = [deploywebappstack-deploymentfrontendbucket67ceb713-bdwdhoteyaww](url)
DeployWebAppStack.deploymentCloudFrontURL05BF422F = [d1d9mac06h49pc.cloudfront.net](url)

Konstrictorman commented:
Infra folder is empty !

Where are these scripts: Added npm scripts for building the app, synthesizing the CloudFormation template, and deploying. ?

Saidmuhammad2002 commented on infra/package.json:
```diff
@@ -0,0 +1,29 @@
+    "cdk": "cdk",
+    "build:app": "npm --prefix ../ run build:infra",
+    "synthesize": "cdk synth > MyStack.template.yaml",
+    "deploy": "npm run build:app && npm run synthesize && cdk deploy"
```
Single script for deploying and all other proccess related to that

Saidmuhammad2002 commented:
> Infra folder is empty !
> 
> Where are these scripts: Added npm scripts for building the app, synthesizing the CloudFormation template, and deploying. ?

Hi sorry somehow my infra folder not added i updated and now you can see infra folder and in package.json you can find scripts you mentioned
